src/core/positioning/bluetoothreceiver.cpp

```diff
@@ -163,43 +163,37 @@ void BluetoothReceiver::doConnectDevice()
 
 void BluetoothReceiver::setSocketState( const QBluetoothSocket::SocketState socketState )
 {
-  if ( mSocketState == static_cast<QAbstractSocket::SocketState>( socketState ) )
-  {
-    return;
-  }
+  emit socketStateChanged( static_cast<QAbstractSocket::SocketState>( socketState ) );
+  emit socketStateStringChanged( socketStateString() );
+}
+
+QAbstractSocket::SocketState BluetoothReceiver::socketState()
+{
+  return mSocket ? static_cast<QAbstractSocket::SocketState>( mSocket->state() ) : QAbstractSocket::UnconnectedState;
```

Comment:
```suggestion
  if ( mSocket == nullptr )
  {
    return QAbstractSocket::UnconnectedState;
  }
  
 return static_cast<QAbstractSocket::SocketState>( mSocket->state() );
```

not all will agree but i found that more readable

in fact mSocket could be not a pointer then there would be no need to check his validity